pkg/storage/storage_tests/storage_test.go

```diff
@@ -0,0 +1,124 @@
+package storagetest
+
+// The goal of this package is to provide common testing ground for all
+// storage tests which can be extracted to achieve more DRY code.
+// This package should contain black box tests such as Save-Get roundtrip,
+// Error checking, etc.
+// More detailed tests verifying internal behavior should be still implemented
+// as part of specific storage packages.
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"testing"
+
+	"github.com/gobuffalo/suite"
+	"github.com/gomods/athens/pkg/storage"
+	"github.com/gomods/athens/pkg/storage/fs"
+	"github.com/gomods/athens/pkg/storage/mem"
+	"github.com/gomods/athens/pkg/storage/minio"
+	"github.com/gomods/athens/pkg/storage/mongo"
+	"github.com/gomods/athens/pkg/storage/rdbms"
+)
+
+type TestSuites struct {
+	*suite.Model
+	storages []storage.TestSuite
+	module   string
+	version  string
+	mod      []byte
+	zip      []byte
+	info     []byte
+}
+
+func (d *TestSuites) SetupTest() {
+	ra := d.Require()
+
+	//
+	fsTests, err := fs.NewTestSuite(d.Model)
+	ra.NoError(err)
+	d.storages = append(d.storages, fsTests)
+
+	// mem
+	memStore, err := mem.NewTestSuite(d.Model)
+	ra.NoError(err)
+	d.storages = append(d.storages, memStore)
+
+	// minio
+	minioStorage, err := minio.NewTestSuite(d.Model)
+	ra.NoError(err)
+	d.storages = append(d.storages, minioStorage)
+
+	// mongo
+	mongoStore, err := mongo.NewTestSuite(d.Model)
+	ra.NoError(err)
+	d.storages = append(d.storages, mongoStore)
+
+	// rdbms
+	rdbmsStore, err := rdbms.NewTestSuite(d.Model)
+	d.Model.SetupTest()
+	d.storages = append(d.storages, rdbmsStore)
```

Comment:
can we put each of these tests runs into its own package?

for example, in the `disk` package:

```go
func TestDiskStorages(t *testing.T) {
    diskSuite, err := NewTestSuite(d.Model)
    if err != nil {
        t.Fatal(err)
    }
    tst := storagetests.Test(diskSuite)
    tst.SetupTest()
    tst.Run()
}
```

Reply:
i create module_storage package
another one would be cdn_storage, up to separate PR, i will create issue for this once this is in